api/v1beta1/nmstate_types.go

```diff
@@ -30,15 +31,22 @@ type NMStateSpec struct {
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 }
 
+// NMStateStatus defines the observed state of NMState
+type NMStateStatus struct {
+	Conditions shared.ConditionList `json:"conditions,omitempty"`
+}
+
 // +kubebuilder:object:root=true
 // +kubebuilder:resource:path=nmstates,scope=Cluster
 // +kubebuilder:storageversion
+// +kubebuilder:subresource=status
 
 // NMState is the Schema for the nmstates API
 type NMState struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-	Spec              NMStateSpec `json:"spec,omitempty"`
+	Spec              NMStateSpec   `json:"spec,omitempty"`
+	Status            NMStateStatus `json:"status,omitempty"`
```

Comment:
I don't see the controller code that update the status.

Reply:
The controller doesn't do anything with that. OLM uses the Conditions for it's own operator.